### ISSUE TITLE
feat(audio): add the Inkling dMel audio front end (log-mel extractor, 16-bin quantizer, summed bin embeddings)

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1311]
## Summary

Inkling consumes audio as discretized mel frames ("dMel"): 80 log-mel channels at 20 frames per second, each channel quantized to one of 16 bins, embedded through a single `[80 * 16, hidden]` table, summed across channels and RMS-normed into one soft token per 50 ms frame. This issue adds the feature extractor (host side), the quantizer, the tower, placeholder expansion of `<|unused_200053|>`, and the merge, so `mlxcel generate --audio` and `input_audio` chat parts work on #1318.

## Current behavior

`src/audio/feature_extractor.rs:139` (`AudioFeatureExtractor`) and `src/audio/whisper_mel.rs:200` (`log_mel_spectrogram`) implement Whisper-style 25 ms / 10 ms log-mel features with natural-log compression; nothing produces 100 ms / 50 ms `log10` features, slaney-normalized filters, or bin indices, and no tower sums per-channel embeddings. `src/loading/mod.rs` has no audio-capable Inkling loader.

## Expected behavior

Audio config (`audio_config`): `model_type: "inkling_audio"`, `n_mel_bins: 80`, `mel_vocab_size: 16`, `dmel_min_value: -7.0`, `dmel_max_value: 2.0`, `rms_norm_eps: 1e-6`, `decoder_dmodel` / `text_hidden_size` = text hidden, `max_frames_per_chunk: 256` (tower chunking only). `processor_config.json`: `feature_extractor = {feature_size: 80, sampling_rate: 16000, hop_length: 800, n_fft: 1600, window_size: 1600, audio_token_duration_s: 0.05, window_size_multiplier: 2.0, padding_value: 0.0, return_attention_mask: true}`, `num_dmel_bins: 16`, `audio_token: "<|unused_200053|>"`, `audio_bos_token: "<|content_audio_input|>"`.

Feature extraction (per clip, mono f32 at 16 kHz; stereo is averaged; other rates are rejected, so resample on the host first with the existing `resample_to_16k`):

```
hop = round(0.05 * 16000) = 800;  win = round(0.05 * 2.0 * 16000) = 1600;  n_fft = win
window = periodic Hann of length win (zero-padded symmetrically to n_fft when win < n_fft)
mel_filters [80, n_fft/2 + 1 = 801]: slaney mel scale and slaney normalization (librosa-style, f_min = 0, f_max = 8000), computed in f64 then cast
padded_length = longest clip in the batch (or max_length with truncation)
right_pad = (-padded_length) mod hop;  left_pad = n_fft - hop = 800
signal = zeros(left_pad) ++ clip ++ zeros(padded_length - len + right_pad)
num_frames = 1 + (left_pad + padded_length + right_pad - n_fft) / hop            # == ceil(padded_length / hop)
frame_t = signal[t*hop : t*hop + n_fft] * window;  spectrum = rfft(frame_t)          # center = false
mag = max(|spectrum|, 1e-10)                                                           # [num_frames, 801]
mel = mag @ mel_filters.T;  log_mel = log10(max(mel, 1e-10))                           # [num_frames, 80], f32
valid_frames = ceil(len / hop);  log_mel[valid_frames:] = 0.0;  mask[t] = t < valid_frames
```

(2401 samples give 4 frames, 3 valid.) Accumulate the mel matmul in f32 with per-row dot products (the reference uses a vmap GEMV because a Metal GEMM moved values across nearby bin boundaries).

dMel quantization (per value `v` of `log_mel`):

```
v = clip(v, -7.0, 2.0)
centers = linspace(-7.0, 2.0, 16) (f64);  mid = (centers[:-1] + centers[1:]) / 2        # 15 boundaries
boundary[i] = the largest f32 <= mid[i]   (cast to f32, then nextafter(-inf) where the cast rounded up)
bin = count of i with v > boundary[i]                                                  # 0..15, ties go to the lower bin
audio_input_ids: int32 [frames, 80]
```

Tower (`ids: [F, 80]`): `emb = embed_audio_tokens(ids + arange(80) * 16)` (`[F, 80, hidden]`), `sum` over the 80 channels, `RMSNorm(hidden; norm)` -> `[F, hidden]`; process in chunks of `max_frames_per_chunk` frames. Only valid frames (mask true) are kept, concatenated over clips.

Placeholder expansion: the template renders `<|content_audio_input|><|unused_200053|><|audio_end|>` per audio part; replace the single `<|unused_200053|>` (id 200053) with `valid_frames` copies; the merge writes the frame embeddings at those positions in order (`merge_llava`), after the image merge.

Weights: `model.audio.encoder.weight [1280, hidden]` (MLX 4-bit checkpoints ship it quantized: `.weight` uint32 `[1280, hidden/8]`, `.scales`, `.biases`; load with `UnifiedEmbedding` / `QuantizedEmbedding` and dequantize-on-gather as other quantized embeddings), `model.audio.final_norm.weight [hidden]`. Rename to `audio_tower.embed_audio_tokens.*` and `audio_tower.norm.weight`.

## Implementation plan

1. `src/audio/inkling_dmel.rs`: `InklingFeatureExtractorConfig` (from `processor_config.json` `feature_extractor`, defaults as above), `extract_log_mel(clip: &[f32]) -> (Vec<f32> [F*80], Vec<bool>)` reusing the Hann/rfft helpers in `src/audio/whisper_mel.rs` (add a `log10` variant and a slaney filterbank builder next to the existing mel builder), and `quantize_dmel(log_mel, 16, -7.0, 2.0) -> Vec<i32>` with the boundary rule above.
2. `src/audio/inkling_tower.rs`: `InklingAudioTower { embed: UnifiedEmbedding, norm: RMSNorm, n_mel_bins, mel_vocab_size, chunk }` with `forward(ids: [F, 80] int32) -> [F, hidden]`.
3. Extend the VLM model from #1327 (or, if this lands first, `src/vision/inkling_vl.rs` created here) with `audio_tower` and `audio_token_id`; `get_input_embeddings(ids, pixel_values, audio_ids)` merges images then audio.
4. Wire `--audio` (CLI `src/commands/generate.rs`) and the server `input_audio` content part through `src/multimodal/host_preprocessor.rs` (WAV via `load_wav_file`, resample to 16 kHz, then steps 1 to 2) and expand the placeholder in `src/multimodal/vlm_prompt.rs`.
5. `docs/supported-models.md`: mark audio input.

## Validation

(a) Unit tests, `src/audio/inkling_dmel_tests.rs`:

- `log_mel_matches_numpy_reference_on_random_noise`: 2401 random samples give `[4, 80]`, frames 0..3 valid, values within 2e-6 of a straightforward f64 STFT + slaney filterbank computed in the test.
- `frame_count_and_mask_follow_hop`: 16000 samples -> 20 frames, all valid; 16001 -> 21 frames, 21 valid; batched `[16000, 8000]` pads the second clip to 20 frames with 10 valid.
- `dmel_boundaries_round_down_and_ties_go_low`: the 15 boundaries are each `<=` the f64 midpoint; `v == boundary` maps to the lower bin; `-7.0 -> 0`, `2.0 -> 15`, `-100 -> 0`, `100 -> 15`.
- `tower_sums_channels_then_norms`: random table, `ids` all zero give `norm(sum_c table[16 c])`.
- `placeholder_expansion_uses_valid_frames` in `src/multimodal/vlm_prompt_tests.rs`.

(b) Real checkpoint: `models/Inkling-Small-mlx-4bit` (`mlx-community/Inkling-Small-mlx-4bit`, about 153 GB):

```bash
./target/release/mlxcel generate -m models/Inkling-Small-mlx-4bit --audio tests/fixtures/speech_16k.wav -p "Transcribe the audio." -n 64
```

Acceptance: the transcript matches the fixture's spoken sentence (word error rate below 20 percent on a clean 5 second clip); the number of audio placeholder tokens equals `ceil(samples / 800)`; on `models/Inkling-0.6B-A0.6B` the `audio_input_ids` for the fixture are identical to the reference processor's output and the tower output matches within 1e-3 relative.

## Acceptance criteria

- [ ] Feature extractor, quantizer and tower follow the formulas above and are unit-tested against f64 references
- [ ] An audio request on Inkling-Small 4-bit yields a correct transcription through the CLI and `/v1/chat/completions`
- [ ] docs/supported-models.md updated
- [ ] detection table in src/models/detection.rs updated with a test
- [ ] cargo test --workspace --profile test-fast --features metal,accelerate passes
- [ ] cargo clippy --workspace --all-targets -- -D warnings and cargo fmt --all -- --check pass

## Out of scope

- Audio output (Inkling is text-out only).
- Non-WAV decoding beyond what `src/audio/preprocessing_*.rs` already handles.

## Dependencies

- Blocked by #1318.

- Part of epic #1313
